### PR TITLE
ci: release: test migrations before building

### PR DESCRIPTION
**Pull request by johnstcn**
- Adds step to test migrations before releasing.
- Modifies `test-migrations` target to avoid using `migrate-ci`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -785,10 +785,12 @@ test-postgres: test-postgres-docker
 
 test-migrations: test-postgres-docker
 	echo "--- test migrations"
+	set -euxo pipefail
 	COMMIT_FROM=$(shell git rev-parse --short HEAD)
 	COMMIT_TO=$(shell git rev-parse --short main)
-	DB_NAME=$(shell go run scripts/migrate-ci/main.go)
-	go run ./scripts/migrate-test/main.go --from="$$COMMIT_FROM" --to="$$COMMIT_TO" --postgres-url="postgresql://postgres:postgres@localhost:5432/$$DB_NAME?sslmode=disable"
+	REF=$(shell git rev-parse --short HEAD)
```
Comment (deansheather) [inline Makefile:791]: Why run the same command twice?

**Review by deansheather — COMMENTED** [review]
Does this work when testing cherry pick releases?

File: Makefile (hunk 1 above)
Comment (johnstcn) [reply Makefile:791]: Cause I have the dumb

**Comment by johnstcn** [comment]
> Does this work when testing cherry pick releases?

The script stands up a test database, runs the migrations from `$COMMIT_FROM`, runs the migrations from `$COMMIT_TO`, dumps the schema, and diffs it against `$COMMIT_TO:dump.sql`.
I don't think it should make any difference between a cherry-picked commit or not.

File: Makefile [hunk 2]
```diff
@@ -785,10 +785,11 @@ test-postgres: test-postgres-docker
 
 test-migrations: test-postgres-docker
 	echo "--- test migrations"
+	set -euxo pipefail
```
Comment (mtojek) [inline Makefile:788]: Is this safe here? We won't leak any CI secrets accidentally?

**Review by deansheather — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (johnstcn) [reply Makefile:788]: I'll remove the `-x` just to be sure.

**Review by mtojek — APPROVED** [review]

**Comment by mtojek** [comment]
Related: https://github.com/coder/coder/issues/13008

Can we merge it now and resolve the issue?

**Pull request merged**